samples/extensions/mesh_shader_culling/README.md

```diff
@@ -0,0 +1,198 @@
+
+## Task Shader
+
+A task shader is an optional stage responsible for launching mesh shaders.  It has two purposes:
+
+* Decide how many mesh shaders to launch in the workgroup.
+* Create a task payload that mesh shaders will have read-only access to.
+  * taskPayloadSharedEXT type variable in GLSL can exist at most once it gets implicitly used by EmitMeshTasksEXT 
+    and behaves like shared memory in the task shader and like a read-only SSBO in a mesh shader.
+
+In general, one should use task shaders anytime you use the mesh pipeline.  While it is "optional,"  their use is 
+strongly encouraged to get the most out of the mesh shading pipeline.  The only time one wouldn't use them is for 
+very simplistic scenes such as rendering a single triangle found in the [mesh shader sample](../mesh_shading).
+
+```glsl
+// Example of the data shared with its associated mesh shader:
+// 1) define some structure if more than one variable data sharing is desired:
+// Please note: GPU vendors recommend to use as little task payload as possible, eg. by packing the data to fewer bits etc.
+struct SharedData
+{
+    vec4  positionTransformation;
+    int   N;
+    int   meshletsNumber;
+    float subDimension;
+    float cullRadius;
+};
+// 2) use the following variable with a storage class specifier to "establish the connection"
+taskPayloadSharedEXT SharedData sharedData;
+```
+
+Once a variable is defined with taskPayloadSharedEXT in the task shader, it will be shared with the mesh shader when 
+EmitMeshTasksEXT is called.
+
+In general, the Mesh pipeline refers to a new pipeline 
+which replaces everything before the fragment shader 
+with an (optional) task shading stage that can call other mesh 
+shading stages.
+
+* A task shader (optional) is used to launch mesh shader workgroup(s)
+* A mesh shader has the responsibility to generate 
+  primitives and vertices.
+
+More details about emitting a mesh task can be found in the attached article:
+
+[Mesh Shading For Vulkan](https://www.khronos.org/blog/mesh-shading-for-vulkan)
+[Introduction to Turing Mesh Shaders](https://developer.nvidia.com/blog/introduction-turing-mesh-shaders/)
+
+GPU manufactures have recommended best practices for 
+their hardware in setting the work group and mesh size 
+number. Further reading can be found here:
+
+* [Meshlet Size tradeoffs](https://zeux.io/2023/01/16/meshlet-size-tradeoffs/)
+* NVIDIA - [Mesh Shaders in Turing](https://on-demand.gputechconf.com/gtc-eu/2018/pdf/e8515-mesh-shaders-in-turing.pdf)
+* AMD - [Sampler feedback ultimate in Mesh shaders](https://gpuopen.com/wp-content/uploads/slides/AMD_RDNA2_DirectX12_Ultimate_SamplerFeedbackMeshShaders.pdf)
+* [Timur's blog](https://timur.hu/blog/2022/mesh-and-task-shaders)
+
+## Mesh Shader
+
+A mesh shader does not execute on the same workgroup as the task shader that emitted it if a task shader exists.  
+Many workgroups can be  One task shader workgroup can launch many mesh shader workgroups.
```

Comment:
I think this paragraph still needs editing. Suggestion:

Task and mesh shaders are executed in workgroups similar to compute shaders. Each task shader workgroup can launch many mesh shader workgroups. Each mesh shader workgroup is responsible for generating vertices and primitives. The API allows for any logic, but a typical application it is recommended to organize this around meshlets, which are a small group of vertices and primitives. Typically, each task shader invocation processes a group of meshlets and each mesh shader workgroup processes one meshlet.